wallet/controllers.go

```diff
@@ -43,8 +43,7 @@ func (service *Service) LookupPublicKey(ctx context.Context, keyID string) (*htt
 
 func validateHTTPSignature(ctx context.Context, r *http.Request, signature string) (string, error) {
 	// validate that the signature in the header is valid based on public key provided
-	var s httpsignature.Signature
-	err := s.UnmarshalText([]byte(signature))
+	s, err := httpsignature.SignatureParamsFromRequest(r)
```

Comment:
Hopefully the right layer: should we validate `SignatureParamsFromRequest` length? Im not sure off the top of my head if there is a max acceptable length? Or if that is even a concern (I cant image much security risk)

Reply:
Added an explicit check for size = 0